src/const_eval.md

```diff
@@ -42,7 +42,8 @@ r[const-eval.const-expr.path-item]
   Recursively defining constants is not allowed.
 
 r[const-eval.const-expr.path-static]
-* Paths to [statics]. These are only allowed within the initializer of a static.
+* Paths to [statics] with these restrictions and observations.
+  * In particular, reads and writes to any `static mut` or [`extern` statics] is not allowed.
```

Comment:
So over in #1657, there is proposed language:

> Static initializers may refer to and even read from other statics.
> When reading from mutable statics, they read the initial value of that static.

The language in this PR seems contradictory to the language in that one, no?

Reply:
Yeah this already works on stable
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=59b79563db4fe5d5df26ab3f27cd4f1c

Reply:
@RalfJung Can you write down exactly what we should say here? It would be helpful since we don't know exactly what the rules are. From looking at the implementation, I'm guessing, is it the `CanAccessMutGlobal` setting, which appears to be set only when in a static initializer? Does this need to say that you cannot read from `static mut` unless inside a static initializer, and that you can never write to a `static mut`, and that reading *and* writing `extern` statics always disallowed?

Reply:
Yes:
- reads from mutable statics (`static mut`, and interior mutable `static`) are forbidden exception inside the initializer expression of a `static`/`static mut`
- writes to mutable statics are forbidden everywhere
- reads from and writes to extern static are forbidden everywhere

Reply:
I have updated the text just now, so that the cases for the two evaluation contexts are clearer.